packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/ClusterManagersController.java

```diff
@@ -89,11 +102,28 @@ void addClusterManagers(@NonNull List<Messages.PlatformClusterManager> clusterMa
   void addClusterManager(String clusterManagerId) {
     ClusterManager<MarkerBuilder> clusterManager =
         new ClusterManager<MarkerBuilder>(context, googleMap, markerManager);
-    ClusterRenderer<MarkerBuilder> clusterRenderer =
-        new ClusterRenderer<MarkerBuilder>(context, googleMap, clusterManager, this);
+    initializeRenderer(clusterManager);
+    clusterManagerIdToManager.put(clusterManagerId, clusterManager);
+  }
+
+  /**
+   * Initialize cluster renderer if it hasn't been initialized before. ClusterManager uses default
+   * renderer which should be changed to (legacy) renderer or advanced renderer
+   */
+  private void initializeRenderer(ClusterManager<MarkerBuilder> clusterManager) {
+    final ClusterRenderer<MarkerBuilder> renderer = clusterManager.getRenderer();
+    if (renderer.getClass() == MarkerClusterRenderer.class
```

Comment:
Why do we need this reflection here?  Can we use non null instead as a signal here?

Reply:
I've removed this check - previously `initializeRenderer` (it was named differently) was called multiple times, now it's called once and there's no need to check if a renderer has been previously assigned